scripts/delayed.js

```diff
@@ -1,7 +1,36 @@
 // eslint-disable-next-line import/no-cycle
 import { sampleRUM } from './aem.js';
+import { getConfigValue } from './configs.js';
 
 // Core Web Vitals RUM collection
 sampleRUM('cwv');
 
 // add more delayed functionality here
+
+// Load Commerce events SDK and collector
+const config = {
+  environmentId: await getConfigValue('commerce-environment-id'),
+  environment: await getConfigValue('commerce-environment'),
+  storeUrl: await getConfigValue('commerce-store-url'),
+  websiteId: await getConfigValue('commerce-website-id'),
+  websiteCode: await getConfigValue('commerce-website-code'),
+  storeId: await getConfigValue('commerce-store-id'),
+  storeCode: await getConfigValue('commerce-store-code'),
+  storeViewId: await getConfigValue('commerce-store-view-id'),
+  storeViewCode: await getConfigValue('commerce-store-view-code'),
+  websiteName: await getConfigValue('commerce-website-name'),
+  storeName: await getConfigValue('commerce-store-name'),
+  storeViewName: await getConfigValue('commerce-store-view-name'),
+  baseCurrencyCode: await getConfigValue('commerce-base-currency-code'),
+  storeViewCurrencyCode: await getConfigValue('commerce-base-currency-code'),
+  storefrontTemplate: 'Franklin',
+};
+
+window.adobeDataLayer.push(
+  { storefrontInstanceContext: config },
+  { eventForwardingContext: { commerce: true, aep: false } },
+);
+
+// Load events SDK and collector
+import('./commerce-events-sdk.js');
+import('./commerce-events-collector.js');
```

Comment:
@herzog31  Since this is all delayed, does it mean that any event occurring after 3s will not be collected ? E.g. if the user clicks away or closes the window before this is loaded ?

Reply:
If a user leaves the page before this is loaded, the events will not be collected. That's correct and unfortunately a known limitation.

There are some ideas to fix this in https://git.corp.adobe.com/mabecker/acdl2 which can persist events in session storage to make sure they can be collected on the next page. But this will take some more work and will be added to the [mainline ACDL](https://github.com/adobe/adobe-client-data-layer) in the near future.